src/context.rs

```diff
@@ -158,10 +158,15 @@ fn build_partition_columns(
     partition_stats: &Statistics,
     schema: SchemaRef,
 ) -> Vec<PartitionColumn> {
-    // TODO PhysicalPartitionColumn might not be the right data structure here (lacks ID etc)
+    // TODO PartitionColumn might not be the right data structure here (lacks ID etc)
     match &partition_stats.column_statistics {
+        // NB: Here we may end up with `null_count` being None, but DF pruning algorithm demands that
+        // the null count field be not nullable itself, and consequently for any such cases the
+        // pruning will fail, and we will default to using all partitions.
         Some(column_statistics) => zip(column_statistics, schema.fields())
             .map(|(stats, column)| {
+                // TODO: the to_string will discard the timezone for Timestamp* values, and will
+                // therefore hinder the ability to recreate them once needed for partition pruning
```

Comment:
Yeah, need to think about it some more.

Reply:
Actually, since DF [doesn't even collect](https://github.com/apache/arrow-datafusion/blob/5ddad472acbbe2073aa6ac901ee5115f5830cc0f/datafusion/core/src/datasource/file_format/parquet.rs#L193-L355) statistics for timestamp types, for now this problem won't manifest itself (but we also won't be able to prune by timestamp-like columns).